### PR TITLE
toybox

### PR DESCRIPTION
**Pull request by asdfugil**

### CODE REVIEW

File: makefiles/toybox.mk [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	$(SED) -i 's|syscall(__NR_|syscall(|g' $(BUILD_WORK)/toybox/toys/other/ionice.c
+	$(SED) -i 's|UT_LINESIZE|13838852|g' $(BUILD_WORK)/toybox/toys/pending/last.c
+	$(SED) -i '1 i\#define LOGIN_NAME_MAX 256' $(BUILD_WORK)/toybox/toys/pending/useradd.c
+	$(CP) -a $(BUILD_MISC)/toybox/config $(BUILD_WORK)/toybox/.config
+
+ifneq ($(wildcard $(BUILD_WORK)/toybox/.build_complete),)
+toybox:
+	@echo "Using previously built toybox."
+else
+toybox: toybox-setup openssl libxcrypt
```
Comment (Diatrus) [inline makefiles/toybox.mk:24]: libxcrypt won't build on macOS.

File: build_info/toybox.control [hunk 2]
```diff
@@ -0,0 +1,18 @@
+Package: toybox
+Version: @DEB_TOYBOX_V@
+Author: Rob Landley <rob@landley.net>
+Maintainer: @DEB_MAINTAINER@
+Depends: libcrypt2, libssl1.1, libiosexec1
```
Comment (Diatrus) [inline build_info/toybox.control:5]: No libcrypt or libiosexec on macOS

**Comment by CRKatri** [comment]
You need `toybox.control` and `toybox.control.macosx`, if you don’t have `toybox.control` targets which don’t have a platform specific one will fail.

**Comment by asdfugil** [comment]
all right fixed

**Comment by asdfugil** [comment]
aight this *should* work on rootless

**Review by Gymxo — COMMENTED** [review]
Don't procursus have openssl3?

File: makefiles/toybox.mk [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+toybox: toybox-setup openssl $(TOYBOX_DEPS)
+	$(MAKE) -C $(BUILD_WORK)/toybox
+	$(INSTALL) -m755 $(BUILD_WORK)/toybox/toybox $(BUILD_STAGE)/toybox/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/bin
+	$(call AFTER_BUILD)
+endif
+
+toybox-package: toybox-stage
+	# toybox.mk Package Structure
+	rm -rf $(BUILD_DIST)/toybox
+	mkdir -p $(BUILD_DIST)/toybox/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/bin
```
Comment (Gymxo) [inline makefiles/toybox.mk:42]: Remove this

**Comment by CRKatri** [comment]
Thanks! This is gonna really come in handy!

**Pull request merged**